backend/rs/canisters/ic_vetkeys_encrypted_maps_canister/tests/tests.rs

```diff
@@ -154,6 +159,707 @@ fn map_sharing_should_work() {
+
+        // let all_maps = env.query::<Vec<EncryptedMapData<AccessRights>>>(
+        //     user,
+        //     "get_all_accessible_encrypted_maps",
+        //     encode_one(()).unwrap(),
+        // );
+
+        // assert_eq!(
+        //     all_values,
+        //     all_maps
+        //         .iter()
+        //         .map(|m| (
+        //             (
+        //                 m.map_owner,
+        //                 Blob::<32>::try_from(m.map_name.as_ref()).unwrap()
+        //             ),
+        //             m.keyvals
+        //                 .iter()
+        //                 .map(|(map_key, value)| (
+        //                     Blob::<32>::try_from(map_key.as_ref()).unwrap(),
+        //                     value.clone()
+        //                 ))
+        //                 .collect::<Vec<_>>()
+        //         ))
+        //         .collect::<Vec<_>>()
+        // );
+
+        let all_destructured = all_values.into_iter().next().unwrap();
+        assert_eq!((caller, map_name.clone()), all_destructured.0);
+        let computed_map_wildcard = all_destructured.1;
+        assert_eq!(
+            expected_map.clone(),
+            BTreeMap::from_iter(computed_map_wildcard)
+        );
+
+        // for map in all_maps {
+        //     if access_rights.can_get_user_rights() {
+        //         assert_eq!(
+        //             BTreeMap::<Principal, AccessRights>::from_iter(
+        //                 map.access_control
+        //                     .into_iter()
+        //                     .chain(std::iter::once((caller, access_rights)))
+        //             ),
+        //             BTreeMap::from_iter(authorized_users.clone().into_iter())
+        //         );
+        //     } else {
+        //         assert_eq!(map.access_control, vec![]);
+        //     }
+        // }
+    }
+}
+
+#[test]
+fn should_modify_key_value_in_map() {
+    let rng = &mut reproducible_rng();
+    let env = TestEnvironment::new(rng);
+    let caller = random_self_authenticating_principal(rng);
+    let map_name = random_map_name(rng);
+    let map_key = random_map_key(rng);
+    let encrypted_value = random_encrypted_value(rng);
+
+    env.update::<Result<Option<ByteBuf>, String>>(
+        caller,
+        "insert_encrypted_value",
+        encode_args((
+            caller,
+            map_name.clone(),
+            map_key.clone(),
+            encrypted_value.clone(),
+        ))
+        .unwrap(),
+    )
+    .unwrap();
+
+    let new_encrypted_value = random_encrypted_value(rng);
+    assert_eq!(
+        env.update::<Result<Option<ByteBuf>, String>>(
+            caller,
+            "insert_encrypted_value",
+            encode_args((
+                caller,
+                map_name.clone(),
+                map_key.clone(),
+                new_encrypted_value
+            ))
+            .unwrap(),
+        ),
+        Ok(Some(encrypted_value))
+    );
```

Comment:
To be sure that the _change_ was written, should we also assert that `new_encrypted_value` is returned when just reading the value?

Reply:
Will be addressed in [CRP-2884](https://dfinity.atlassian.net/browse/CRP-2884)